src/relations/deprecated_shared_db_database_provides.py

```diff
@@ -0,0 +1,340 @@
+
+import logging
+import typing
+
+import ops
+
+import mysql_shell
+import relations.remote_databag as remote_databag
+import status_exception
+
+if typing.TYPE_CHECKING:
+    import abstract_charm
+
+
+class LogPrefix(logging.LoggerAdapter):
+    """Add prefix to all log messages"""
+
+    def process(self, msg, kwargs):
+        return f"[DEPRECATED shared-db] {msg}", kwargs
+
+
+logger = LogPrefix(logging.getLogger(__name__), extra=None)
+
+
+class _RemoteUnitDatabag(remote_databag.RemoteDatabag):
+    """Relation databag for remote unit"""
+
+    def __init__(self, relation: ops.Relation) -> None:
+        self._app_name = relation.app.name
+        self._endpoint_name = relation.name
+        if not relation.units:
+            raise remote_databag.IncompleteDatabag(
+                app_name=self._app_name, endpoint_name=self._endpoint_name
+            )
+        # Subordinate charms can only access principal unit's databag
+        assert len(relation.units) == 1
+        # Principal unit
+        remote_unit = relation.units.copy().pop()
+        dict.__init__(self, relation.data[remote_unit])
+
+
+class _RelationBreaking(Exception):
+    """Relation will be broken after the current event is handled"""
+
+
+class _Relation:
+    """Relation to one application charm"""
+
+    def __init__(
+        self,
+        *,
+        relation: ops.Relation,
+        peer_relation_app_databag: ops.RelationDataContent,
+    ) -> None:
+        self._id = relation.id
+        self._peer_app_databag = peer_relation_app_databag
+        self._peer_databag_username_key = f"deprecated_shared_db_relation_{self._id}.username"
+        self.peer_databag_password_key = f"deprecated_shared_db_relation_{self._id}.password"
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, _Relation):
+            return False
+        return self._id == other._id
+
+
+class _UnitThatNeedsUser(_Relation):
+    """Related application unit that has needs user password"""
+
+    def __init__(
+        self,
+        *,
+        relation: ops.Relation,
+        peer_relation_app_databag: ops.RelationDataContent,
+        unit: ops.Unit,
+    ) -> None:
+        super().__init__(relation=relation, peer_relation_app_databag=peer_relation_app_databag)
+        self._local_unit_databag = relation.data[unit]
+        self._remote_unit_databag = _RemoteUnitDatabag(relation)
+        assert len(relation.units) == 1
+        self._remote_unit_name = relation.units.copy().pop().name
+
+    def set_databag(
+        self,
+        *,
+        password: str,
+    ) -> None:
+        """Share connection information with application charm."""
+        logger.debug(f"Setting unit databag {self._id=} {self._remote_unit_name=}")
+        self._local_unit_databag["allowed_units"] = self._remote_unit_name
+        self._local_unit_databag["db_host"] = "127.0.0.1"
+        self._local_unit_databag["db_port"] = "3306"
+        self._local_unit_databag["wait_timeout"] = "3600"
```

Comment:
So, this comes from [mysql configuration](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_wait_timeout)

The legacy charm supports a config and default it to 3600, but since we are not supporting the config, we should reflect the sysvar default, i.e. 28800.
A task was added to mirror this on the vm charm.